docs/tutorials/skweak.ipynb

```diff
@@ -0,0 +1,10139 @@
+      "545fe54bdb8748e59af035eb280f373e",
+      "e6d7f804c97a42caa3f79d253760526c",
+      "b46f71e18c464b50bbbc3148215a69e9",
+      "c9f539140ecd44cb96ff995a5408e1c5",
+      "2ef53978ca784d1895b78bcf76fa47da",
+      "193765beb3bb4931a02eab5f7a6cb7d8",
+      "7246a59eb1ad43a796f25a0f449afe94",
+      "7537c122f57b43468432d75ec99a4800",
+      "172f2a03aecb45d4875aab13f56218a4",
+      "a488937c83c54842bde74ce8f189252a",
+      "9b8d4afa26634a9d8638ef1539ef1eaa",
+      "8e84f6d946284a2cb39c7a493884d411",
+      "41162d35bd394f44bc45a5cfdc6b12e6",
+      "fcd29046b1334c319c810c316b19dc0f",
+      "2dffe7f1cbd445bea8168a3f44f8ef05",
+      "d435ac65b4094d668d7b492a6fb8a9b8",
+      "0512ff2effab46568d60afbf11829cf2",
+      "ba0a0c30e911425dacba47a25e10c714",
+      "0f0ab519c6e844ca94be1a38443fe969"
+     ]
+    },
+    "id": "e75a815f",
+    "outputId": "17b4893d-ba33-4645-9f07-bed5c513cf64"
+   },
+   "outputs": [],
+   "source": [
+    "tag_set = {'O': 0, 'B-PER': 1, 'I-PER': 2, 'B-ORG': 3, 'I-ORG': 4, 'B-LOC': 5, 'I-LOC': 6, 'B-MISC': 7, 'I-MISC': 8}\n",
+    "tag_set = { v:k for k,v in tag_set.items() }\n",
+    "\n",
+    "def convert_ner_tags(record, tag_set=None):\n",
+    "    record['ner_tags'] = [ tag_set[x] for x in record['ner_tags'] ]\n",
+    "    return record\n",
+    "\n",
+    "def strip_BI_tags(record):\n",
+    "    record['ner_tags'] = [ x.lstrip('B-').lstrip('I-') for x in record['ner_tags'] ]\n",
+    "    return record\n",
+    "\n",
+    "dataset = dataset\\\n",
+    "    .map(convert_ner_tags, fn_kwargs={\"tag_set\": tag_set})\\\n",
+    "    .map(strip_BI_tags)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e075c5e",
+   "metadata": {
+    "id": "9e075c5e"
+   },
+   "source": [
+    "We will now convert the training and validation splits of our dataset into spaCy Doc objects. \n",
+    "\n",
+    "spaCy demands strings to be given as inputs to a tokenizer. However, as our dataset is already tokenized, we bypass this restriction by using our own tokenizer and encapsulating our tokens in a class that inherits from `str`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "80b62499",
+   "metadata": {
+    "id": "80b62499"
+   },
+   "outputs": [],
+   "source": [
+    "import spacy\n",
+    "from spacy.tokens import Doc\n",
+    "from dataclasses import dataclass\n",
+    "\n",
+    "@dataclass\n",
+    "class Record(str):\n",
+    "    tokens: list\n",
+    "\n",
+    "def custom_tokenizer(text):\n",
+    "    return Doc(nlp.vocab, text.tokens)\n",
+    "\n",
+    "nlp = spacy.load(\"en_core_web_sm\", disable=[\"ner\", \"lemmatizer\"])\n",
+    "nlp.tokenizer = custom_tokenizer\n",
+    "\n",
+    "training_set = [ Record(x) for x in dataset[\"train\"][\"tokens\"] ]\n",
+    "dev_set = [ Record(x) for x in dataset[\"validation\"][\"tokens\"] ]\n",
+    "\n",
+    "train_docs = list(nlp.pipe(training_set))\n",
+    "dev_docs = list(nlp.pipe(dev_set))"
```

Comment:
Do we need to use `nlp.pipe`? Maybe we can directly create `Doc` instances here?

Reply:
I tried directly creating `Doc` instances but it triggered an error message with `skweak`, so I went with `nlp.pipe`.

`nlp.pipe` is also used in the tutorials on the `skweak` repository.